src/cadenza.js

```diff
@@ -502,6 +572,42 @@ function validGeometryType(value) {
   ].includes(value);
 }
 
+/** @param {string} [value] */
+function validMediaType(value) {
+  return value === CSV || value === MS_EXCEL_2007;
+}
+
+/** @param {Promise<Response>} responsePromise
+ * @param {object} [options] - Options
+ * @param {string} [options.filename] - The media type to use for the data
+ * */
+async function download(responsePromise, { filename } = {}) {
+  const res = await responsePromise;
+  assert(res.ok, 'download failed');
```

Comment:
why do you here throw a generic error instead of a CadenzaError?

Reply:
Should be handled in #fetch already, see my comment there.

Reply:
Fixed